### PR TITLE
use external perl file

### PR DESCRIPTION
**Pull request by andreyorst**
Moved the Perl code to snippets.pl file

### CODE REVIEW

**Comment by andreyorst** [comment]
It's based on auto discard, since it's the most up to date branch

File: snippets.pl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if ($placeholder_id eq "0") {
+        $placeholder_id = "9999";
+    }
+    $placeholder_ids[$i] = $placeholder_id;
+    if (defined($placeholder_default)) {
+        $placeholder_id_to_default{$placeholder_id} = $placeholder_default;
+    }
+}
+
+print("evaluate-commands -client $client %{ set-option window snippets_placeholder_info");
```
[the remark this answers is not in the document]
Comment (andreyorst) [reply snippets.pl]: Oh, for some reason I thought that perl runs in the background here.

File: snippets.pl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    print(" $placeholder_id");
+    if (exists $placeholder_id_to_default{$placeholder_id}) {
+        print("|1");
+    } else {
+        print("|0");
+    }
+}
+print("}\n");
+
+print("evaluate-commands -client $client %{ set-register dquote");
```
[the remark this answers is not in the document]
Comment (andreyorst) [reply snippets.pl]: Changed that

**Comment by occivink** [comment]
Thanks, I'll try to get back on the auto-discard branch soon

**Pull request merged**